src/aks-preview/azext_aks_preview/managed_cluster_decorator.py

```diff
@@ -2276,6 +2276,60 @@ def get_nodepool_taints(self) -> Union[List[str], None]:
         """
         return self.agentpool_context.get_node_taints()
 
+    def _get_enable_cost_analysis(self, enable_validation: bool = False) -> bool:
+        """Internal function to obtain the value of enable_cost_analysis.
+
+        When enabled, if both enable_cost_analysis and disable_cost_analysis are
+        specified, raise a MutuallyExclusiveArgumentError.
+
+        :return: bool
+        """
+        enable_cost_analysis = self.raw_param.get("enable_cost_analysis")
+
+        # This parameter does not need dynamic completion.
+        if enable_validation:
+            if enable_cost_analysis and self._get_disable_cost_analysis(enable_validation=False):
+                raise MutuallyExclusiveArgumentError(
+                    "Cannot specify --enable-cost-analysis and --disable-cost-analysis at the same time."
+                )
+
+        return enable_cost_analysis
+
+    def get_enable_cost_analysis(self) -> bool:
+        """Obtain the value of enable_cost_analysis.
+
+        :return: bool
+        """
+        return self._get_enable_cost_analysis(enable_validation=True)
+
+    def _get_disable_cost_analysis(self, enable_validation: bool = False) -> bool:
+        """Internal function to obtain the value of disable_vpa.
+
+        When enabled, if both enable_cost_analysis and disable_cost_analysis are
+        specified, raise a MutuallyExclusiveArgumentError.
+
+        :return: bool
+        """
+        # Read the original value passed by the command.
+        disable_cost_analysis = self.raw_param.get("disable_cost_analysis")
+
+        # This option is not supported in create mode, hence we do not read the property value from the `mc` object.
+        # This parameter does not need dynamic completion.
+        if enable_validation:
+            if disable_cost_analysis and self._get_enable_cost_analysis(enable_validation=False):
+                raise MutuallyExclusiveArgumentError(
+                    "Cannot specify --enable-cost-analysis and --disable-cost-analysis at the same time."
+                )
```

Comment:
It seems odd that you have this check in both directions - is this needed to handle both orders of the options? (I'm assuming this is a pattern used with other settings as well.)

Reply:
I copied this pattern from disable_vpa but agree it is sorta weird we check in both directions. Let me confirm I don't need both and if so I'll remove one.

Reply:
yeah only one direction is needed. Removed the other one